core/src/test/java/jenkins/util/VirtualFileTest.java

```diff
@@ -1000,6 +1001,7 @@ public void testCanRead_True_FileVF() throws Exception {
     }
 
     @Test
+    @Ignore("TODO doesn't pass on ci.jenkins.io due to root user being used in container tests")
```

Comment:
feel free to amend this with assume or whatever, all I have time for now

Reply:
I opened [INFRA-2879](https://issues.jenkins.io/browse/INFRA-2879) to keep track of what needs to be done here. In short, we build jnlp-inbound image for every major tool, like maven,python,node,etc... and for each time we never configure the user jenkins:jenkins.